config/config.go

```diff
@@ -165,3 +189,36 @@ func (c JobConfig) ScrapeInterval() time.Duration {
 func (c JobConfig) ScrapeTimeout() time.Duration {
 	return stringToDuration(c.GetScrapeTimeout())
 }
+
+// TargetGroups returns the encapsulated target groups with sorted targets.
+func (c JobConfig) TargetGroups() (groups []*TargetGroup) {
+	for _, tg := range c.GetTargetGroup() {
+		g := &TargetGroup{
+			Labels: make(clientmodel.LabelSet),
+		}
+		for _, pair := range tg.GetLabels().GetLabel() {
+			g.Labels[clientmodel.LabelName(pair.GetName())] = clientmodel.LabelValue(pair.GetValue())
+		}
+		for _, t := range tg.GetTarget() {
+			g.Targets = append(g.Targets, clientmodel.LabelSet{"__host__": clientmodel.LabelValue(t)})
```

Comment:
So far we manage all special-purpose label names in the client library (like `clientmodel.MetricNameLabel`). Probably we should do the same here.